components/support/nimbus-fml/src/editing/cursor_position.rs

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::ops::Add;
+
+use unicode_segmentation::UnicodeSegmentation;
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct CursorPosition {
+    pub line: u32,
+    pub col: u32,
+}
+
+#[allow(dead_code)]
+impl CursorPosition {
+    pub(crate) fn new(line: usize, col: usize) -> Self {
+        Self {
+            line: line as u32,
+            col: col as u32,
+        }
+    }
+}
+
+/// CursorSpan is used to for reporting errors and defining corrections.
+/// This is passed across the FFI and used by the editor.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct CursorSpan {
+    pub from: CursorPosition,
+    pub to: CursorPosition,
+}
+
+/// CursorPosition + &str -> CursorSpan.
+/// This is a convenient way of making CursorSpans.
+impl Add<&str> for CursorPosition {
+    type Output = CursorSpan;
+
+    fn add(self, rhs: &str) -> Self::Output {
```

Comment:
What does `rhs` stand for? Right-hand side? Sorry it just wasn't clear to me.

Reply:
Yes.